### PR TITLE
Adding clusterScoped flag to deploy only cluster scoped resources

### PR DESCRIPTION
**Pull request by eletonia**
The `clusterScoped` flag is required to deploy M4D on the Operate First cluster using ArgoCD. This PR adds checks to each cluster-scoped resource Helm template, and will only deploy them if `clusterScoped` is `true`.

### CODE REVIEW

**Review by roee88 — CHANGES_REQUESTED** [review]
Can you please change the logic to deploy these cluster scoped resources if clusterScoped is true? 

That is, treat it as a boolean and also reverse the logic (to match the variable name).

**Comment by eletonia** [comment]
Ok I reversed the logic and am treating `clusterScoped` as a boolean now

**Review by roee88 — COMMENTED** [review]
There are too rbac files in katalog directory that also need this flag check

File: charts/m4d/templates/manager-auth-proxy-service.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{- if include "m4d.isEnabled" (tuple .Values.manager.enabled (or .Values.coordinator.enabled .Values.worker.enabled)) }}
+{{- if .Values.clusterScoped}}
```
[the remark this answers is not in the document]
Comment (eletonia) [reply charts/m4d/templates/manager-auth-proxy-service.yaml:2]: Oh if services are not cluster scoped, I can remove this from this file. I just assumed everything with `manager-auth-proxy-*` is cluster scoped.

**Comment by eletonia** [comment]
> There are two rbac files in katalog directory that also need this flag check

Which files are these?

**Comment by roee88** [comment]
> > There are two rbac files in katalog directory that also need this flag check
> 
> Which files are these?

https://github.com/IBM/the-mesh-for-data/blob/master/charts/m4d/templates/katalog-connector/katalog-connector-rbac.yaml

https://github.com/IBM/the-mesh-for-data/blob/master/charts/m4d/templates/katalog-connector/katalog-rbac.yaml

**Review by roee88 — APPROVED** [review]

**Review by roee88 — APPROVED** [review]

**Comment by roee88** [comment]
Thanks!

**Pull request merged**